### DISCUSSION TITLE
Download progress of remote assets is always 0

### DISCUSSION THREAD

**Discussion by CraftGamesDevelop** [jelte/be.khepri.play.assetdelivery.addressables#42] [Q&A]
I check download percentage like this but it is always 0

`var asyncTask = Khepri.AssetDelivery.AddressablesAssetDelivery.GetDownloadSizeAsync(sceneByName.Value);
			while (!asyncTask.IsDone) yield return null;
			isCached = (asyncTask.Status == UnityEngine.ResourceManagement.AsyncOperations.AsyncOperationStatus.Succeeded && asyncTask.Result == 0);
			if (!isCachedDep)
			{
				
				_requestDependens = Addressables.DownloadDependenciesAsync(sceneByName.Value);
				
				while (_requestDependens.Status != AsyncOperationStatus.Succeeded)
				{
					sizeFile.Value = asyncTask.Result / 1000000f;
					progress.Value = _requestDependens.PercentComplete; // progress
					yield return null;
				}`

**Comment by jelte** [comment]
hey,
to check the download progress:

https://github.com/jelte/be.khepri.play.assetdelivery.addressables/discussions/17#discussioncomment-166896

or do you mean that `_requestDependens.PercentComplete;` is always 0?

**Reply by CraftGamesDevelop** [reply]
Yes, I mean: _requestDependens.PercentComplete; is always 0

**Comment by OlesiaPashko** [comment]
I have the save issue with _requestDependens.PercentComplete; is always 0. Could you please help?

**Comment by phanphantz** [comment]
I have the same issue. Also tried using GetDownloadStatus() instead of PercentComplete, same result.

IEnumerator Start()
{
        var handle = Addressables.LoadAssetAsync<GameObject>("VideoPlayer");
        while (!handle.IsDone)
        {       
            text.text = handle.GetDownloadStatus().Percent + "% (" + handle.GetDownloadStatus().DownloadedBytes + " B)";
            yield return null;
        }
        var result = Instantiate(handle.Result);
        result.transform.SetParent(transform);
        result.transform.localScale = Vector3.one;
        result.transform.localPosition = Vector3.zero;
}

It's always display 0 for both Percent and DownloadedBytes.

**Comment by phanphantz** [comment]
@jelte @CraftGamesDevelop @OlesiaPashko Hello, all. I just found the solution to this issue and created [a pull request](https://github.com/jelte/be.khepri.play.assetdelivery.addressables/pull/72) to the repo.

In **WebRequestAssetBundleResourceHandlerBase.cs** I set the progress report callback for **m_ProvideHandle** so that the download progress in **handle.PercentComplete** will be reported by **webRequest.downloadProgress**

You guys can check out my fork while the pull request is pending.
https://github.com/phanphantz/be.khepri.play.assetdelivery.addressables

**EDIT:** Looks like the current fix only work when using **LoadAssetAsync()** not **LoadAssetsAsync()**. Will investigate further.
